libc-top-half/musl/src/thread/sem_timedwait.c

```diff
@@ -8,7 +8,9 @@ static void cleanup(void *p)
 
 int sem_timedwait(sem_t *restrict sem, const struct timespec *restrict at)
 {
+#ifdef __wasilibc_unmodified_upstream
 	pthread_testcancel();
+#endif	
```

Comment:
`src/thread/pthread_testcancel.c` is super trivial so maybe easier to just include that now rather than add ifdef here.

See https://github.com/WebAssembly/wasi-libc/blob/main/libc-top-half/musl/src/thread/pthread_testcancel.c